mono/mini/interp/interp.c

```diff
@@ -170,7 +170,7 @@ frame_data_allocator_alloc (FrameDataAllocator *stack, InterpFrame *frame, int s
 		/* First allocation by this frame. Save the markers for restore */
 		if (infos_len == stack->infos_capacity) {
 			stack->infos_capacity = infos_len * 2;
-			stack->infos = g_realloc (stack->infos, stack->infos_capacity * sizeof (FrameDataInfo));
+			stack->infos = (FrameDataInfo*)g_realloc (stack->infos, stack->infos_capacity * sizeof (FrameDataInfo));
 		}
```

Comment:
Don't we have c++ template wrappers for these so these casts are not necessary ?

Reply:
I see there is this g_cast. Not sure why the warnings are present. Compiled normally mono/mono on ubuntu (with gcc). Well that would explain. We compile with C compiler, so template is not present, but `-Wc++-compat` is passed.